src/main/java/com/hermesworld/ais/galapagos/topics/service/impl/ValidatingTopicServiceImpl.java

```diff
@@ -263,6 +264,31 @@ public CompletableFuture<Void> changeTopicOwner(String environmentId, String top
                 .orElseGet(() -> topicService.changeTopicOwner(environmentId, topicName, newApplicationOwnerId));
     }
 
+    @Override
+    public Optional<TopicMetadata> getSingleTopic(String environmentId, String topicName) {
+        Optional<TopicMetadata> optionalMetadata = getTopic(environmentId, topicName);
+
+        if (optionalMetadata.isEmpty()) {
+            return Optional.empty();
+        }
+
+        TopicMetadata metadata = optionalMetadata.get();
+        String ownerApplicationId = metadata.getOwnerApplicationId();
+
+        boolean hasAccess = applicationsService.getUserApplications().stream().map(KnownApplication::getId)
```

Comment:
I don't think you have to check user's access here. The other methods of ValidatingTopicServiceImpl also do not do that. Also, **every** authenticated user is allowed to see a topic, not only the owner.